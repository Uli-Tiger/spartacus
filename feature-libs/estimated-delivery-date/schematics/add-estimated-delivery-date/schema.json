{
  "$schema": "http://json-schema.org/schema",
  "$id": "EstimatedDeliveryDateSchematics",
  "title": "Estimated Delivery Date Schematics",
  "type": "object",
  "properties": {
    "project": {
      "type": "string",
      "description": "The name of the project.",
      "$default": {
        "$source": "projectName"
      }
    },
    "debug": {
      "description": "Display additional details during the running process.",
      "type": "boolean",
      "default": false
    },
    "lazy": {
      "type": "boolean",
      "description": "Lazy load the Estimated Delivery Date.",
      "default": true
    },
    "features": {
      "type": "array",
      "uniqueItems": true,
      "default": ["Estimated-Delivery-Date"]
    }
  },
  "required": []
}
