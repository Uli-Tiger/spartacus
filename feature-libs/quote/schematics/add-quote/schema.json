{
  "$schema": "http://json-schema.org/schema",
  "$id": "QuoteSchematics",
  "title": "Quote Schematics",
  "type": "object",
  "properties": {
    "project": {
      "type": "string",
      "description": "The name of the project.",
      "$default": {
        "$source": "projectName"
      }
    },
    "debug": {
      "description": "Display additional details during the running process.",
      "type": "boolean",
      "default": false
    },
    "lazy": {
      "type": "boolean",
      "description": "Lazy load the quote features.",
      "default": true
    },
    "features": {
      "type": "array",
      "uniqueItems": true,
      "default": ["Quote"]
    }
  },
  "required": []
}
