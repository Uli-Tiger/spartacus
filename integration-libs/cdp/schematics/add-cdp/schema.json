{
  "$schema": "http://json-schema.org/schema",
  "$id": "CDPSchematics",
  "title": "CDP Schematics",
  "type": "object",
  "properties": {
    "project": {
      "type": "string",
      "description": "The name of the project.",
      "$default": {
        "$source": "projectName"
      }
    },
    "debug": {
      "description": "Display additional details during the running process.",
      "type": "boolean",
      "default": false
    },
    "lazy": {
      "type": "boolean",
      "description": "Lazy load the CDP feature.",
      "default": true
    },
    "features": {
      "type": "array",
      "uniqueItems": true,
      "default": ["CDP"]
    }
  },
  "required": []
}
