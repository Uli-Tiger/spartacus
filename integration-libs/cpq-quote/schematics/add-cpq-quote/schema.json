{
  "$schema": "http://json-schema.org/schema",
  "$id": "CpqQuoteSchematics",
  "title": "cpq-quote Schematics",
  "type": "object",
  "properties": {
    "project": {
      "type": "string",
      "description": "The name of the project.",
      "$default": {
        "$source": "projectName"
      }
    },
    "debug": {
      "description": "Display additional details during the running process.",
      "type": "boolean",
      "default": false
    },
    "lazy": {
      "type": "boolean",
      "description": "Lazy load the cpq-quote.",
      "default": true
    },
    "features": {
      "type": "array",
      "uniqueItems": true,
      "default": ["cpq-quote"]
    }
  },
  "required": []
}
