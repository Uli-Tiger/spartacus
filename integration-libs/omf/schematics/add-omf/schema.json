{
  "$schema": "http://json-schema.org/schema",
  "$id": "OmfSchematics",
  "title": "OMF Schematics",
  "type": "object",
  "properties": {
    "project": {
      "type": "string",
      "description": "The name of the project.",
      "$default": {
        "$source": "projectName"
      }
    },
    "debug": {
      "description": "Display additional details during the running process.",
      "type": "boolean",
      "default": false
    },
    "features": {
      "type": "array",
      "uniqueItems": true,
      "default": ["OMF"]
    }
  },
  "required": []
}
