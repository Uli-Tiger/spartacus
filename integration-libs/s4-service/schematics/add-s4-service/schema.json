{
  "$schema": "http://json-schema.org/schema",
  "$id": "S4ServiceSchematics",
  "title": "S/4HANA Service Integration Schematics",
  "type": "object",
  "properties": {
    "project": {
      "type": "string",
      "description": "The name of the project.",
      "$default": {
        "$source": "projectName"
      }
    },
    "debug": {
      "description": "Display additional details during the running process.",
      "type": "boolean",
      "default": false
    },
    "lazy": {
      "type": "boolean",
      "description": "Lazy load the S/4HANA Service Integration feature.",
      "default": true
    },
    "features": {
      "type": "array",
      "uniqueItems": true,
      "default": ["s4-service"]
    }
  },
  "required": []
}
