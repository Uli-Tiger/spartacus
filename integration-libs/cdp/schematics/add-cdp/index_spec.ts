/// <reference types="jest" />

import {
  SchematicTestRunner,
  UnitTestTree,
} from '@angular-devkit/schematics/testing';
import {
  Schema as ApplicationOptions,
  Style,
} from '@schematics/angular/application/schema';
import { Schema as WorkspaceOptions } from '@schematics/angular/workspace/schema';
import {
  CDP_FEATURE_NAME,
  SPARTACUS_SCHEMATICS,
  SPARTACUS_CDP,
  LibraryOptions as CdpOptions,
  SpartacusOptions,
  CUSTOMER_TICKETING_FEATURE_NAME,
  cdpFeatureModulePath,
  customerTicketingFeatureModulePath,
  customerTicketingWrapperModulePath,
} from '@spartacus/schematics';
import * as path from 'path';

const collectionPath = path.join(__dirname, '../collection.json');

describe('Spartacus CDP schematics: ng-add', () => {
  const schematicRunner = new SchematicTestRunner(
    SPARTACUS_CDP,
    collectionPath
  );

  let appTree: UnitTestTree;

  const workspaceOptions: WorkspaceOptions = {
    name: 'workspace',
    version: '0.5.0',
  };

  const appOptions: ApplicationOptions = {
    name: 'schematics-test',
    inlineStyle: false,
    inlineTemplate: false,
    routing: false,
    style: Style.Scss,
    skipTests: false,
    projectRoot: '',
    standalone: false,
  };

  const spartacusDefaultOptions: SpartacusOptions = {
    project: 'schematics-test',
    lazy: true,
    features: [],
  };

  const libraryNoFeaturesOptions: CdpOptions = {
    project: 'schematics-test',
    lazy: false,
    features: [],
  };

  const cdpFeatureOptions: CdpOptions = {
    ...libraryNoFeaturesOptions,
    features: [CDP_FEATURE_NAME],
  };

  beforeEach(async () => {
    schematicRunner.registerCollection(
      SPARTACUS_SCHEMATICS,
      path.join(
        __dirname,
        '../../../../projects/schematics/src/collection.json'
      )
    );

    appTree = await schematicRunner.runExternalSchematic(
      '@schematics/angular',
      'workspace',
      workspaceOptions
    );

    appTree = await schematicRunner.runExternalSchematic(
      '@schematics/angular',
      'application',
      appOptions,
      appTree
    );

    appTree = await schematicRunner.runExternalSchematic(
      SPARTACUS_SCHEMATICS,
      'ng-add',
      { ...spartacusDefaultOptions, name: 'schematics-test' },
      appTree
    );
  });

  describe('Without CDP features', () => {
    beforeEach(async () => {
      appTree = await schematicRunner.runSchematic(
        'ng-add',
        libraryNoFeaturesOptions,
        appTree
      );
    });

    it('should not create any of the feature modules', () => {
      expect(appTree.exists(cdpFeatureModulePath)).toBeFalsy();
    });
  });

  describe('With CDP feature', () => {
    describe('general setup', () => {
      beforeEach(async () => {
        appTree = await schematicRunner.runSchematic(
          'ng-add',
          {
            ...cdpFeatureOptions,
            features: [CUSTOMER_TICKETING_FEATURE_NAME],
          },
          appTree
        );
        appTree = await schematicRunner.runSchematic(
          'ng-add',
          cdpFeatureOptions,
          appTree
        );
      });

      it('should install necessary Spartacus libraries', () => {
        const packageJson = JSON.parse(appTree.readContent('package.json'));
        let dependencies: Record<string, string> = {};
        dependencies = { ...packageJson.dependencies };
        dependencies = { ...dependencies, ...packageJson.devDependencies };

        for (const toAdd in dependencies) {
          if (
            !dependencies.hasOwnProperty(toAdd) ||
            toAdd === SPARTACUS_SCHEMATICS
          ) {
            continue;
          }
          const expectedVersion = (dependencies as Record<string, string>)[
            toAdd
          ];
          const expectedDependency = dependencies[toAdd];
          expect(expectedDependency).toBeTruthy();
          expect(expectedDependency).toEqual(expectedVersion);
        }
      });

      it('should add the feature using the lazy loading syntax', async () => {
        const module = appTree.readContent(cdpFeatureModulePath);
        expect(module).toMatchSnapshot();
      });
      it('should install the appropriate dependencies', async () => {
        const customerTicketingWrapperModule = appTree.readContent(
          customerTicketingWrapperModulePath
        );
        expect(customerTicketingWrapperModule).toMatchSnapshot();
      });
    });

    describe('eager loading', () => {
      beforeEach(async () => {
        appTree = await schematicRunner.runSchematic(
          'ng-add',
          {
            ...cdpFeatureOptions,
            features: [CUSTOMER_TICKETING_FEATURE_NAME],
          },
          appTree
        );
        appTree = await schematicRunner.runSchematic(
          'ng-add',
          { ...cdpFeatureOptions, lazy: false },
          appTree
        );
      });

      it('should import appropriate modules', async () => {
        const module = appTree.readContent(cdpFeatureModulePath);
        expect(module).toMatchSnapshot();
        expect(
          appTree.readContent(customerTicketingFeatureModulePath)
        ).toBeTruthy();
      });
    });
  });
});
