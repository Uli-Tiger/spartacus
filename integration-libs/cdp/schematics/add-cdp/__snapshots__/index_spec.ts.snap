// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spartacus CDP schematics: ng-add With CDP feature eager loading should import appropriate modules 1`] = `
"import { NgModule } from '@angular/core';
import { CdpCustomerTicketingModule } from "@spartacus/cdp/customer-ticketing";

@NgModule({
  declarations: [],
  imports: [
    CdpCustomerTicketingModule
  ]
})
export class CdpFeatureModule { }
"
`;

exports[`Spartacus CDP schematics: ng-add With CDP feature general setup should add the feature using the lazy loading syntax 1`] = `
"import { NgModule } from '@angular/core';
import { CdpCustomerTicketingModule } from "@spartacus/cdp/customer-ticketing";

@NgModule({
  declarations: [],
  imports: [
    CdpCustomerTicketingModule
  ]
})
export class CdpFeatureModule { }
"
`;

exports[`Spartacus CDP schematics: ng-add With CDP feature general setup should install the appropriate dependencies 1`] = `""`;
